modules/cimg/src/cimgutils.cpp

```diff
@@ -606,20 +606,31 @@ TIFFHeader getTIFFHeader(const std::string& filename) {
     TIFFSetDirectory(tif, 0);
 
     uint16 samplesPerPixel = 1, bitsPerSample = 8, sampleFormat = 1;
-    TIFFGetFieldDefaulted(tif, TIFFTAG_BITSPERSAMPLE, &bitsPerSample);
-    TIFFGetFieldDefaulted(tif, TIFFTAG_SAMPLESPERPIXEL, &samplesPerPixel);
-    TIFFGetFieldDefaulted(tif, TIFFTAG_SAMPLEFORMAT, &sampleFormat);
-
-    std::array<unsigned int, 2> xres = {150, 1};
-    std::array<unsigned int, 2> yres = {150, 1};
-    // X and Y resolution tags are stored as RATIONAL, i.e. a fractional value
-    // (2 unsigned int values with the first being the numerator, the second the denominator)
-    TIFFGetFieldDefaulted(tif, TIFFTAG_XRESOLUTION, xres.data());
-    TIFFGetFieldDefaulted(tif, TIFFTAG_YRESOLUTION, yres.data());
-    const dvec2 res{xres[0] / static_cast<double>(xres[1]), yres[0] / static_cast<double>(yres[1])};
+    if (!TIFFGetFieldDefaulted(tif, TIFFTAG_BITSPERSAMPLE, &bitsPerSample)) {
+        bitsPerSample = 8;
+    }
+    if (!TIFFGetFieldDefaulted(tif, TIFFTAG_SAMPLESPERPIXEL, &samplesPerPixel)) {
+        samplesPerPixel = 1;
+    }
+    if (!TIFFGetFieldDefaulted(tif, TIFFTAG_SAMPLEFORMAT, &sampleFormat)) {
+        sampleFormat = 1;
+    }
+
+    float xres = 1.0f;
+    float yres = 1.0f;
+    // X and Y resolution tags are stored as float
+    if (!TIFFGetFieldDefaulted(tif, TIFFTAG_XRESOLUTION, &xres)) {
+        xres = 1.0f;
```

Comment:
I guess using 0 would break stuff?

Reply:
Yes, the "physical" extent or print size of an image is obtained by dividing the image width / height by the matching resolution (dpi or dots per cm).